cypress/support/commands.js

```diff
@@ -23,3 +23,235 @@
 //
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
+
+/**
+ * Sets the Dashboard page data range
+ * @param {String} start Start datetime to set
+ * @param {String} end  End datetime to set
+ */
+Cypress.Commands.add('setDashboardDataRange', (start, end) => {
+  cy.get('[data-test-subj="superDatePickerShowDatesButton"]').should('be.visible').click()
+
+  cy.get('[data-test-subj="superDatePickerendDatePopoverButton"]').should('be.visible').click()
+  cy.get('[data-test-subj="superDatePickerAbsoluteTab"]').should('be.visible').click()
+  cy.get('[data-test-subj="superDatePickerAbsoluteDateInput"]').should('be.visible').type('{selectall}' + start)
+  cy.get('[data-test-subj="superDatePickerendDatePopoverButton"]').should('be.visible').click()
+  cy.get('[data-test-subj="superDatePickerAbsoluteTab"]').should('not.exist')
+
+  cy.get('[data-test-subj="superDatePickerstartDatePopoverButton"]').should('be.visible').click()
+  cy.get('[data-test-subj="superDatePickerAbsoluteTab"]').should('be.visible').click()
+  cy.get('[data-test-subj="superDatePickerAbsoluteDateInput"]').should('be.visible').type('{selectall}' + end)
+  cy.get('[data-test-subj="superDatePickerstartDatePopoverButton"]').should('be.visible').click()
+
+  cy.get('[data-test-subj="querySubmitButton"]').should('be.visible').click()
+})
+
+/**
+ * Add saved Dashboard panels
+ * @param {String} keyword Search term for panels of interest
+ * @param {String} type Panel type to search for
+ * @param {Boolean} multiplePages Whether there are multiple pages to iterate through
+ */
+Cypress.Commands.add('addDashboardPanels', (keyword, type, multiplePages = true) => {
+  const iteratePages = () => {
+    cy.get('[data-test-subj="pagination-button-next"]').then(($nextBtn) => {
+      cy.get('[data-test-subj="savedObjectFinderItemList"]').should('be.visible')
+      if ($nextBtn.is(':enabled')) {
+        cy.wrap($nextBtn).click()
+        cy.get('[data-test-subj^="savedObjectTitle' + keyword.replace(/\s+/g, '-') + '"]').should('be.visible').each(($button) => {
+          cy.wrap($button).click()
+        })
+        iteratePages()
+      }
```

Comment:
Just making sure, does this iteratePages call happen after all the panels are added?

Reply:
Yes, the recursive iteratePages call happens after all panels on the current page are added.